app/src/main/java/com/android/streetworkapp/model/park/ParkViewModel.kt

```diff
@@ -80,6 +81,18 @@ open class ParkViewModel(private val repository: ParkRepository) : ViewModel() {
    */
   fun createPark(park: Park) = viewModelScope.launch { repository.createPark(park) }
 
+  /**
+   * Get or create a park by its location.
+   *
+   * @param location The park location.
+   */
+  fun getOrCreateParkByLocation(location: ParkLocation) {
+    viewModelScope.launch {
+      val park = repository.getOrCreateParkByLocation(location)
+      _park.value = park
+    }
```

Comment:
Comment: after some test on my side, this function work as expected. There is unfortunately a bug on the first time that this function is used if there is no park already on the database: The overview screen will show the n-1 park, and not the current n park. To resolve this I created a function :

fun setDefaultPark() {
    getOrCreateParkByLocation(ParkLocation(0.0,0.0,"default"))
  }

in ParkViewModel that is used at the launch of the MapUI. This solves the bug and the map works as intended. You might find it useful for the implementation in the MapUi. You can already define the setDefaultPark() function in this PR (that is my advice) or solve this problem in the next PR.

Reply:
After testing on my side, I observed the same problem. But since the viewmodel itself is functional and correctly working, I think it's up to the integration to adapt to the viewmodel. So I'm leaving this modification to [this](https://github.com/SwEnt-Group8/Street-work-app/pull/56) PR, does that work for you?

Reply:
I think this is a good way to handle this problem. Thank you!